backend/Backend.cpp

```diff
@@ -166,6 +206,97 @@ std::tuple<int, int> Backend::GetExtraClientRange(
   return std::make_tuple(client_start, client_size);
 }
 
```

Comment:
Please help add some description for this new GetExtraClientRange2 function to help others understanding.